docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,594 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">if not np.any(cand):
    return None
</pre><p>I think we should <code>break</code> rather than returning <code>None</code> . This way, we end up always returning a numpy array (which may be empty in some cases). In general, we should avoid mixing the return type (i.e., <code>None</code>  vs <code>ndarray</code> ) and always try to keep the same return type.</p><p> </p><p>If we were to be consistent with the suggestions and variable name changes in Phase 1 above, then Phase 2 might be:</p><p> </p><pre class="ql-syntax" spellcheck="false">def _prune_candidates(subseqs, min_dist, is_cands):
&nbsp; &nbsp; n, m = subseqs.shape
&nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; is_cands = np.zeros(n, dtype=bool)
&nbsp; &nbsp; cands_idx = np.flatnonzero(is_cands)
&nbsp; &nbsp; excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
&nbsp; &nbsp; r = m * (1.0 - ((min_dist ** 2.0) / (2.0 * m)))&nbsp;

&nbsp; &nbsp; for i in range(0, n):
&nbsp; &nbsp; &nbsp; &nbsp; if not np.any(is_cands):
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; break
&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; non_trivial_cands = cands_idx[cands_idx &gt; i + excl_zone]&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; if not len(non_trivial_cands):
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; break
&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; R = np.matmul(subseqs[non_trivial_cands], subseqs[i])
&nbsp; &nbsp; &nbsp; &nbsp; mask = np.flatnonzero(R &gt; r)
&nbsp; &nbsp; &nbsp; &nbsp; is_cands[non_trivial_cands[mask]] = False
&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; return is_cands
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Reply:
Thanks for the input! Didn't know about that. Will be fixed in next commits.